src/Components/App/App.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState, FC } from 'react';
+import './App.css';
+import Header from '../Header/Header'
+import Form from '../Form/Form'
+import HomeCityCards from '../HomeCityCards/HomeCityCards'
+import { fetchCityForecast } from '../../apiCalls'
+import { WeatherLocation } from "../../model/Weather";
+
+const App: FC = () => {
+  const [cities, setCities] = useState<WeatherLocation[]>([]);
+  const [error, setError] = useState('');
+  const [warning, setWarning] = useState('');
+  const [currentCity, setCurrentCity] = useState<WeatherLocation | null>(null);
+  
+  const resetAlerts = () => {
+    setError('');
+    setWarning('');
+  }
+
+  let addCity = async (term: string) => {
+    resetAlerts();
+    const city = await fetchCityForecast(term);
+
+    if (!city) {
+      setError(`No location found called '${term}'`);
+    } else if (cities.find(item => item.id === city.id)) {
+      setWarning(`Location '${term}' is already in the list.`);
+    } else {
+      setCities([city, ...cities]);
+    }
+  };
+  return (
+    <div className="App">
+      <Header />
+      <Form onSearch={addCity}/>
+      {
+        error
```

Comment:
I think to clean this up a bit, we could break the error/warning handlers out to their own competent for the app!